closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -1069,6 +1144,18 @@ private final class EngineAEADDec implements ChaChaEngine {
             tag = new byte[TAG_LENGTH];
         }
 
+        @Override
+        public int getOutputSize(int inLen, boolean isFinal) {
+            // If we are performing a decrypt-update we should always return
+            // zero length since we cannot return any data until the tag has
+            // been consumed and verified.  CipherSpi.engineGetOutputSize will
+            // always set isFinal to true to get the required output buffer
+            // size.
+            return (isFinal ?
+                    Integer.max(Math.addExact((inLen - TAG_LENGTH),
+                            getCipherBufferLength()), 0) : 0);
```

Comment:
This isn't the same as the code it replaces for all values of `inLen`, and the outermost parentheses are unwanted:
```java
            int outLen = 0;
            if (isFinal) {
                outLen = Integer.max(inLen, TAG_LENGTH) - TAG_LENGTH;
                outLen = Math.addExact(outLen, getCipherBufferLength());
            }
            return outLen;
```

Reply:
This code matches what is in the non native ChaCha20 class. Id prefer to keep it such that the classes can be more easily diffed but made this change anyway. If its ok id like to change it back to match the non native class for maintenance reasons.

Yes this is part of what i believe is a bug in our Native Cha Cha code. The non native code determines the output buffer size differently. Matching that code allowed for the failing test to pass without any other unit or openjdk tests failing that i could find.

Reply:
It may be equivalent for real-world values of `inLen` (i.e. `inLen >= Integer.MIN_VALUE + TAG_LENGTH`), but I see no need to settle for an expression that doesn't work for all values.

Reply:
Ok, I think I've convinced myself that part of the problem was that the old code was incorrect; see the [upstream change](https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/cce883c22e8e15ecc1ba7a395f8ca698c781c3f3). We want something equivalent to what you had when you created this pull request.

Reply:
Given we have decided to not make any changes to these lines I believe the code as it stands now has the [original logic present in the ChaChaCipher.java file ](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/fc9b90c93d840c89b162fbf70f508e15616c6fa9/src/java.base/share/classes/com/sun/crypto/provider/ChaCha20Cipher.java#L1541) besides the change in parentheses that we made.